### DISCUSSION TITLE
v0.109.* appears to break path resolution when using uvicorn --root-path "/"

### DISCUSSION THREAD

**Discussion by alexdunnjpl** [fastapi/fastapi#11102] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
full repro delayed pending confirmation that this is not already a known issue

`uvicorn api.app:app --root-path "/" --host 0.0.0.0 --reload`
```

### Description

Upgrade from 0.108.0 to any 0.109.0 version results in 404s for any path (including but not limited to `/docs`) except the root path (`127.0.0.0:8000/`)

Requesting the root path in a browser now resolves to http://127.0.0.1:8000// rather than http://127.0.0.1:8000/

After further investigation it looks like the culprit is the `--root-path` arg provided to uvicorn - previously, setting to `/` was valid/correct, but it appears to no longer be.  It's unclear if this was an intended change or a regression.

Replacing with `--root-path ''` resolves the issue.

### Operating System

Linux

### FastAPI Version

0.109.2

### Pydantic Version

2.0.3

### Python Version

3.10.2

**Comment by alexdunnjpl** [comment]
I've tracked down the issue to uvicorn 0.25.0->0.26.0 and will open an issue there instead.

**Reply by ciaransweet** [reply]
@alexdunnjpl did you get far with this? I've just bumped a repo we have to `0.109.0` and experience the same thing. Was there a version of uvicorn to bump to? 

It appears mine is set to `0.30.4`...

**Reply by alexdunnjpl** [reply]
@ciaransweet looks like [I opened a question here](https://github.com/encode/uvicorn/discussions/2244), but got no response - upvoting it might raise visibility, not sure what their process is.

IIRC, there was no downside to the workaround, so setting the root-path to `''` rather than `'/'` should be fine as a stopgap.
